app/helpers/routing_helper.rb

```diff
@@ -6,54 +6,106 @@ def self.format_seconds_as_minsec(sec)
     format("%<minutes>.2d:%<seconds>.2d", minutes: sec / 60, seconds: sec % 60)
   end
 
-  def self.resolve_coordinates(input)
-    return input if valid_coordinates?(input)
-
-    BuildingMapHelper.destinations[input]
+  def self.calculate_route(start, dest, start_building, dest_building, res)
+    if !start_building[:indoor] && !dest_building[:indoor] # outdoor - outdoor
+      OutdoorRoutingHelper.route_outdoor(start, dest, res)
+    elsif start_building[:indoor]
+      RoutingHelper.handle_start_indoor_cases(dest, start_building, dest_building, res)
+    else
+      OutdoorRoutingHelper.route_outdoor_indoor(start, dest_building, res)
+    end
   end
 
-  def self.valid_coordinates?(coordinates)
-    return false if coordinates.blank?
-
-    regex = /^-?\d{1,2}(\.\d+)?,-?\d{1,2}(\.\d+)?$/
-    coordinates.match(regex)
+  def self.init_routing(params)
+    start = coordinates_from_string(resolve_coordinates(params[:start]))
```

Comment:
Since there now is `start_building`, this should probably be `start_coordinates`, or? (this bothers me more in caller code than here specifically, but it's a general remark)

Reply:
we didn't change this because the call would be too long with that many parameters.